aten/src/ATen/core/ATenDispatch.h

```diff
@@ -103,4 +121,125 @@ class CAFFE2_API ATenDispatch {
 
 CAFFE2_API ATenDispatch& globalATenDispatch();
 
+struct lazy_holder {
+  virtual ~lazy_holder() {}
+};
+
+CAFFE2_API std::list<std::unique_ptr<lazy_holder>>& globalLazyFunctors();
+
+template <bool...>
+struct bool_pack;
+template <bool... bs>
+using all_true = std::is_same<bool_pack<bs..., true>, bool_pack<true, bs...>>;
+
+template <class R, class... Ts>
+using are_all_constructible = all_true<std::is_constructible<R, Ts>::value...>;
```

Comment:
No action needed: this is just for knowledge sharing (I'm adapting this code into the boxed fallback implementation PR, and I just noticed this cleanup).

We have an implementation of c++17 `std::conjunction` in `c10/util/C++17.h`. So you can write this more idiomatically as:

```
template <class R, class... Ts>
using are_all_constructible = std::conjunction<std::is_constructible<R, Ts>...>;
```